### PR TITLE
[flang][hlfir][openacc] Updated LIT tests checks.

### PR DESCRIPTION
**Pull request by vzakhari**
This patch updates the OpenACC LIT tests to invoke HLFIR lowering
in addition to FIR lowering. The tests checks were updated accordingly.

### CODE REVIEW

**Review by clementval — APPROVED** [review]
Thanks for doing this!

File: flang/test/Lower/OpenACC/acc-private.f90 [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
 
-! RUN: bbc -fopenacc -emit-fir %s -o - | FileCheck %s
+! RUN: bbc -fopenacc -emit-fir %s -o - | FileCheck %s --check-prefixes=CHECK,FIR
+! RUN: bbc -fopenacc -emit-hlfir %s -o - | FileCheck %s --check-prefixes=CHECK,HLFIR
 
 ! CHECK-LABEL: acc.firstprivate.recipe @firstprivatization_ref_50xf32 : !fir.ref<!fir.array<50xf32>> init {
 ! CHECK: ^bb0(%{{.*}}: !fir.ref<!fir.array<50xf32>>):
 ! CHECK:   %[[ALLOCA:.*]] = fir.alloca !fir.array<50xf32>
+! FIXME: we need hlfir.declare here to satisfy the assumptions about
+!        the HLFIR lowering, i.e. that every varible has fir/hlfir.declare.
```
Comment (vzakhari) [inline flang/test/Lower/OpenACC/acc-private.f90:10]: Thank you for the review, Valentin!  I wonder if you agree with this note.  I do not think it is a blocker for HLFIR enabling, but I would like to create an issue to track it, if it makes sense.

Comment (clementval) [reply flang/test/Lower/OpenACC/acc-private.f90:10]: Yes I think your note is correct. I can update the recipe generation for that.

Comment (vzakhari) [reply flang/test/Lower/OpenACC/acc-private.f90:10]: Thanks!

**Pull request merged**